paraview/xmls/MergeTreeAutoencoder.xml

```diff
@@ -0,0 +1,629 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ServerManagerConfiguration>
+  <ProxyGroup name="filters">
+    <SourceProxy name="ttkMergeTreeAutoencoder" class="ttkMergeTreeAutoencoder" label="TTK MergeTreeAutoencoder">
+      <Documentation long_help="MergeTreeAutoencoder Long" short_help="MergeTreeAutoencoder Short">This VTK filter uses the ttk::MergeTreeAutoencoder module to compute an auto-encoder of merge trees or persistence diagrams.</Documentation>
+
+      <!-- INPUT DATA OBJECTS -->
+      <InputProperty
+        name="Input"
+        port_index="0"
+        command="SetInputConnection">
+        <ProxyGroupDomain name="groups">
+          <Group name="sources"/>
+          <Group name="filters"/>
+        </ProxyGroupDomain>
+        <DataTypeDomain name="input_type">
+          <DataType value="vtkMultiBlockDataSet"/>
+        </DataTypeDomain>
+        <InputArrayDomain name="input_scalars" number_of_components="1">
+          <Property name="Input" function="FieldDataSelection" />
+        </InputArrayDomain>
+        <Documentation>
+          Merge trees to process.
+        </Documentation>
+      </InputProperty>
+      
+      <InputProperty
+          name="Optional Input"
+          port_index="1"
+          command="SetInputConnection">
+        <ProxyGroupDomain name="groups">
+          <Group name="sources"/>
+          <Group name="filters"/>
+        </ProxyGroupDomain>
+        <DataTypeDomain name="input_type">
+          <DataType value="vtkMultiBlockDataSet"/>
+        </DataTypeDomain>
+        <InputArrayDomain name="input_scalars" number_of_components="1">
+          <Property name="Input" function="FieldDataSelection" />
+        </InputArrayDomain>
+        <Documentation>
+          If input are merge trees, then this input can be used to process join and split trees together. Pass as input either join or split trees in the first input and the other type of trees in the second input.
+          If input are persistence diagrams, then this has no effect to use this input.
+        </Documentation>
+      </InputProperty>
+    
+      <InputProperty
+        name="Info"
+        port_index="2"
+        command="SetInputConnection">
+        <ProxyGroupDomain name="groups">
+          <Group name="sources"/>
+          <Group name="filters"/>
+        </ProxyGroupDomain>
+        <DataTypeDomain name="input_type">
+          <DataType value="vtkTable"/>
+        </DataTypeDomain>
+        <InputArrayDomain name="input_scalars" number_of_components="1">
+          <Property name="Input" function="FieldDataSelection" />
+        </InputArrayDomain>
+        <Documentation>
+          Additionnal information such as cluster assignment for cluster loss (the name of this array must be "ClusterAssignment").
```

Comment:
Done in the next commit!

Reply:
In principle, I guess it'd be better if the visibility of the field "Clustering Array Name" is conditioned by a checkbox "Force Cluster Array", pretty much like the checkbox "Force Input Mask Scalar Field" in `GeometrySmoother`.
If that box is not checked, the code is searching for an array whose name is hard-coded (i.e. `ClusterID`).
Otherwise, if the box is checked, the dropdown menu is shown and the user decides.
That mechanism enables the usage of default values for the selection of arrays.

But that's OK. I guess it's OK like that.
What if the user wants to disable the code associated to the clustering metric? 
Setting the variable "ClusteringLossWeight" to zero is enough to skip all the related code in the execution flow?